Code/Editor/EditorViewportCamera.cpp

```diff
@@ -136,6 +136,43 @@ namespace SandboxEditor
         }
     }
 
+    AZStd::optional<AZ::Transform> CalculateGoToEntityTransform(
+        const AZ::Transform& cameraTransform, const float fovRadians, const AZ::Vector3& center, const float radius)
+    {
+        // do not attempt to interpolate to where we currently are
+        if (cameraTransform.GetTranslation().IsClose(center))
+        {
+            return AZStd::nullopt;
+        }
+
+        const AZ::Vector3 forward = [&center, &cameraTransform]
+        {
+            const AZ::Vector3 forward = (center - cameraTransform.GetTranslation()).GetNormalized();
+            // if the camera is looking directly up or down, pitch the camera down or up respectively to avoid
+            // a singularity when creating the lookat transformation below
+            if (const float forwardDot = forward.Dot(AZ::Vector3::CreateAxisZ()); AZ::IsCloseMag(AZ::Abs(forwardDot), 1.0f, 0.001f))
+            {
+                return AZ::Transform::CreateFromQuaternion(
+                           AZ::Quaternion::CreateFromAxisAngle(cameraTransform.GetBasisX(), AZ::DegToRad(5.0f) * -AZ::GetSign(forwardDot)))
+                    .TransformVector(AZ::Vector3::CreateAxisZ() * AZ::GetSign(forwardDot));
+            }
+            return forward;
+        }();

```

Comment:
What is the benefit of using a lambda here?

Reply:
So the complex initialization of the value can be performed in one place and then `forward` can be made `const` for later use.

One school of thought is to prefer `AZStd::invoke` as opposed to the immediately invoked function expression (IIFE) - more info [here](https://www.cppstories.com/2016/11/iife-for-complex-initialization/), but I prefer the terseness it offers.

e.g.
```c++
const AZ::Vector3 forward = AZStd::invoke([&center, &cameraTransform]
{
    const AZ::Vector3 forward = (center - cameraTransform.GetTranslation()).GetNormalized();
    // if the camera is looking directly up or down, pitch the camera down or up respectively to avoid
    // a singularity when creating the lookat transformation below
    if (const float forwardDot = forward.Dot(AZ::Vector3::CreateAxisZ()); AZ::IsCloseMag(AZ::Abs(forwardDot), 1.0f, 0.001f))
    {
        return AZ::Transform::CreateFromQuaternion(
                   AZ::Quaternion::CreateFromAxisAngle(cameraTransform.GetBasisX(), AZ::DegToRad(5.0f) * -AZ::GetSign(forwardDot)))
            .TransformVector(AZ::Vector3::CreateAxisZ() * AZ::GetSign(forwardDot));
    }
    return forward;
});
```

Reply:
Thanks for explaining it.